src/environments/examples/BitFlippingEnv.jl

```diff
@@ -6,17 +6,43 @@ In Bit Flipping Environment we have n bits. The actions are 1 to n where executi
 Refer [Hindsight Experience Replay paper](https://arxiv.org/pdf/1707.01495.pdf) for the motivation behind the environment.
 """
 
-struct BitFlippingEnv <: AbstractEnv
+mutable struct BitFlippingEnv <: AbstractEnv
     N::Int
     rng::AbstractRNG
     state::BitArray{1}
     goal_state::BitArray{1}
+    max_steps::Int
+    t::Int
+end
+
+function BitFlippingEnv(; N = 8)
+    rng = Random.GLOBAL_RNG
+    state = bitrand(rng, N)
+    goal_state = bitrand(rng, N)
+    max_steps = N
+    BitFlippingEnv(N, rng, state, goal_state, max_steps, 0)
+end
+
+function BitFlippingEnv(; N = 8, T = 8)
+    rng = Random.GLOBAL_RNG
+    state = bitrand(rng, N)
+    goal_state = bitrand(rng, N)
+    max_steps = T
+    BitFlippingEnv(N, rng, state, goal_state, max_steps, 0)
```

Comment:
I think these two methods will be overridden by the next constructor (In Julia, methods are only dispatched by positional arguments).

We can only define one constructor:

```julia
function BitFlippingEnv(; N = 8, T=N, rng = Random.GLOBAL_RNG)
...
end
```